src/dbus/clientprivate.cpp

```diff
@@ -134,11 +146,15 @@ void Ngf::ClientPrivate::ngfdStatus(const QString &service, const QString &arg1,
             QDBusInterface *iface = m_iface;
             m_iface = 0;
             // All currently active events are invalid, so clear event list
-            m_events.clear();
-            // Signal that we are disconnected
-            emit q_ptr->connectionStatus(false);
-            if (iface)
+            while (!m_events.isEmpty()) {
```

Comment:
Same here - just iterate over m_events and `delete` each item, then `.clear()` it.